File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program100.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//		Program:
//			Write a program which accept string from user and display it inn reverse order.			
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{

	private:
		string str;
		int iCnt,iCount;

	public:

		Solution(string str,int iCnt=0,int iCount=0)
		{
			this->str=str;
		}

		void Reverse()
		{
			iCnt=(str.length());

			while(iCnt!=-1)
			{
			
				cout<<str[iCnt];
				iCnt--;	
			}	
			cout<<"\n";
		}
};

int main()
{
	string str;

	cout<<"Enter string: ";
	getline(cin,str);
	
	Solution *sObj = new Solution(str);
	sObj->Reverse();

	delete sObj;
	return 0;

}

/*
OUTPUT:

Enter string: MarvellouS
SuollevraM

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program100.java
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//		Program:
//			Write a program which accept string from user and display it inn reverse order.			
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	String str;
	int iCnt,iCount;

	Solution(String str)
	{
		this.str=str;
		this.iCnt=0;
		this.iCount=0;
	}

	void Reverse()
	{
		
		iCnt = str.length()-1;
		while(iCnt!=-1)
		{
			System.out.print(str.charAt(iCnt));
			iCnt--;
		}
		
		System.out.println();
	}
}

class Program
{
	public static void main(String args[])
	{
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter the string: ");
		String str = sObj.nextLine();

		Solution obj = new Solution(str);
		obj.Reverse();

	}
}

/*
OUTPUT:
Enter the string: 
Vishal
lahsiV

*/
